src/algebra/permutation_matrix.jl

```diff
@@ -0,0 +1,242 @@
+export PermutationMatrix, PT_X_P, PT_X_P!, getind ,getindices
+
+import Base: *
+import LinearAlgebra: transpose , inv, mul!
+
+@doc raw"""
+`PermutationMatrix(ind::Array{T})` creates a permutation matrix with ones at coordinates `(k, ind[k]) for k = 1:length(ind)`.
+
+A permutation matrix represents a matrix containing only zeros and ones, which basically permutes the vector or matrix it is multiplied with.
+These matrices A are constrained by:
+```math
+    A_{ij} = \{0, 1\}\\
+    ∑_{i} A_{ij} = 1\\
+    ∑_{j} A_{ij} = 1
+```
+An example is the 3-dimensional permutation matrix
+```math
+A = \begin{bmatrix} 1 & 0 & 0 \\ 0 & 0 & 1 \\ 1 & 0 & 0 \end{bmatrix}
+```
+"""
+struct PermutationMatrix{ T <: Int } <: AbstractMatrix{T}
```

Comment:
`T <: Int` restricts `T` to be only `Int` because `Int` is a concrete type. Usually `<:` operator is used with abstract types such as `Integer` or `AbstractVector` or `Number`. `T <: Int` does not make a lot of sense in this context as it is equivalent to just `Int`

Reply:
Resolved by 5801d21.